app/views/comments/_form.html.erb

```diff
@@ -66,38 +66,60 @@
 
   <%= javascript_include_tag "dragdrop" %>
   <script>
-    $('#comment-form').bind('ajax:beforeSend', function(event){      
+    $('#comment-form').bind('ajax:beforeSend', function(event){
       $("#text-input").prop('disabled',true)
-      $("#comment-form .btn-primary").button('loading',true);      
+      $("#comment-form .btn-primary").button('loading',true);
     });
-    <% emoji_names = emoji_names_list %>
-    <% emojis = emoji_names[:emojis] %>
-    <% image_map = emoji_names[:image_map] %>
-    var image_map = <%= raw image_map.to_json %>
-    horsey(document.querySelector('textarea#text-input'), {
-        suggestions: (<%= raw emojis.to_json %>),
-        anchor: ':',
-        render: function (li, suggestion) {
-            var image = image_map[suggestion.text];
-            li.innerHTML = "<b>"+image+"</b><span>"+suggestion.text+"</span>";
+
+    var emojis = <%== atwho_emojis %>
```

Comment:
OK, so here is where we are currently loading the emojis, and I believe it's compiled from a static list in this controller line:

https://github.com/publiclab/plots2/pull/2925/files#diff-127a850b7d7b20173ad3fba88f8a25e3R56

What we could do instead is use a `remoteFilter` as we do in the other two atwho configs, but point it at a static file that we create from the Gist I linked to. It could be stored in `/public/emoji.json` and so that would be the route as well. Does that makes sense? Thanks!!!

Reply:
I'm not sure the list I linked to is in the right format either, but it should be relatively easy to parse it.

Reply:
Hi @jywarren  the barnraising event looks very cool, wish I could have attended. I did a project in the UK on  community emergency preparedness and we definitely need people and software to work on this!  

 I've been looking at loading the emoji data from the json file. It's easy enough to just load the emoji characters like so, and they populate the dropdown menu:

```
Option A
      emojis_config = {
        at: ':',
        callbacks: {
          remoteFilter: function(query, callback) {
            $.getJSON("/emoji.json" + query, {}, function(data) {
                callback(Object.values(data).map(function(name) { return name }));
             });
            }
          },
         displayTpl: "<li>${name}</li>",
         insertTpl: "${name}",
         limit: 25
     }
```
But 2 concerns: a) is this OK for the other editor to render the emoji as character? b) can we do something fancier? The at.js demo shows a searchable dropdown with names and emoji images displayed so I know that is possible but perhaps we'd need a different setup, or is it possible to do this with static files? I'd tried reading in the emoji_values.json into the remote filter, then trying to have "data: '/emoji.json' and that displays the emoji names properly, but I'm not sure how you'd display the emoji images in the dropdown via that route:

```
Option B
       emojis_config = {
         at: ':',
         callbacks: {
           remoteFilter: function(query, callback) {
           // this render the names of emojis properly, but we have no "search" functionality
             $.getJSON("/emoji_values.json" + query, {}, function(data) {
                 callback(Object.values(data).map(function(i) { return i.text }));
              });
             }
           },
// unsure how/if we can edit the displayTpl here to display the emoji characters, 
//since those are not in the emoji_json file we read in during callback---maybe if I merged these 2 files?
          displayTpl: "<li>${name}</li>",
          insertTpl: ":${name}:",
         // I don't think this is ever reading from this file but relying on the preexisting gemoji code, when you delete this line, it still renders
          data: '/emoji.json',
          limit: 10
      }
   ``` 

So I could use the Option A above and physically rearrange the json so we get more than just smileys showing in the dropdown, but maybe like 10 most popular emojis or so, then rebase and submit, or try something else. Let me know what you think. Thanks again for your feedback!

Reply:
> for the other editor to render the emoji as character

do you mean like showing the emoji plus the name in the autocomplete, like:

![screen shot 2018-07-19 at 6 00 19 pm](https://user-images.githubusercontent.com/24359/42972484-a00366e4-8b7d-11e8-83bd-9956ec5995cb.png)

? If so, i like the idea of rearranging the JSON... would it be a huge pain? I think i could do it in a script if you want me to try?

Reply:
And i love the idea! But if it's a lot of work we can just do the simple version first and solve that in a follow-up issue... so we can see this go into production first!

Reply:
I think we could use `emoji.json` and make it like:

```
':smile:': "😄 smile",
':smile:': "😄 smile",
':smile:': "😄 smile",
```

(but without repetition, just for example)

Is that what you're thinking?